testsuite/utils.py

```diff
@@ -176,3 +178,26 @@ def check_condition(condition, condition_type, status, reason=None, message=None
     ):
         return True
     return False
+
+
+def is_nxdomain(hostname: str):
+    """
+    Returns True if hostname has no `A` record in DNS. False otherwise.
+    Will raise exception `dns.resolver.NoAnswer` if there exists different record type under the hostname.
+    """
+    try:
+        dns.resolver.resolve(hostname)
+    except dns.resolver.NXDOMAIN:
+        return True
+    return False
+
+
+def sleep_ttl(hostname: str):
+    """Sleeps for duration of TTL for given hostname."""
+    try:
+        response = dns.resolver.resolve(hostname, raise_on_no_answer=False)
+    except dns.resolver.NXDOMAIN:
+        return
+    if response.rrset is None:
+        return
```

Comment:
in what case rrset is none?

Reply:
When DNS server returns empty answer but no error. This usually happens when under the domain name there is different type of record than what was requested in the query. The default behavour of `dns.resolver.resolve` is to query for A record. 

This functions is here mostly for MyPy check failing. I will just add type ignore comment here and remove the `raise_on_no_answer` so if the RRset is empty, exception is thrown.